hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCommitArchiveLog.java

```diff
@@ -103,12 +105,16 @@ private boolean deleteOtherInstants() {
 
         HoodieTable table = HoodieTable.getHoodieTable(new HoodieTableMetaClient(fs, config.getBasePath(), true), config);
         HoodieTimeline commitTimeline = table.getCompletedCommitTimeline();
-        List<String> savepoints = table.getSavepoints();
-
+        // We cannot have any holes in the commit timeline. We cannot archive any commits which are made after the first savepoint present.
+        Optional<HoodieInstant> firstSavepoint = table.getCompletedSavepointTimeline().firstInstant();
         if (!commitTimeline.empty() && commitTimeline.countInstants() > maxCommitsToKeep) {
             // Actually do the commits
-            return commitTimeline.getInstants().filter(s -> !savepoints.contains(s.getTimestamp()))
-                .limit(commitTimeline.countInstants() - minCommitsToKeep);
+            return commitTimeline.getInstants().filter(s -> {
+                // if no savepoint present, then dont filter
```

Comment:
okay. this will retain all commits after the savepoint, until the savepoint is deleted.